### PR TITLE
Initial implementation of the `log` subcommand

### LINKED ISSUE #260: a `log` sub-command
Add a sub-command to the modality probe CLI that turns a report into something `git log` like, with a few options:

* A single probe's timeline.
* A complete timeline, interleaved by probe and/or some other interleaving options.

### PR DESCRIPTION
**Pull request by pittma**
Closes #260.

### CODE REVIEW

File: modality-probe-cli/src/meta.rs [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    pl: Option<u32>,
+) -> Result<Option<String>, Box<dyn std::error::Error>> {
+    match (th, pl) {
+        (Some("i8"), Some(pl)) => Ok(Some(format!("{}", pl as i8))),
+        (Some("i16"), Some(pl)) => Ok(Some(format!("{}", pl as i16))),
+        (Some("i32"), Some(pl)) => Ok(Some(format!("{}", pl as i32))),
+        (Some("u8"), Some(pl)) => Ok(Some(format!("{}", pl as u8))),
+        (Some("u16"), Some(pl)) => Ok(Some(format!("{}", pl as u16))),
+        (Some("u32"), Some(pl)) => Ok(Some(format!("{}", pl as u32))),
+        (Some("f32"), Some(pl)) => Ok(Some(format!("{}", pl as f32))),
```
Comment (jonlamb-gh) [inline modality-probe-cli/src/meta.rs:132]: Should we be doing `f32::from_bits(pl)` here?

File: modality-probe-cli/src/export/mod.rs [hunk 2]
```diff
@@ -51,41 +47,22 @@ pub enum GraphType {
...
-            "failed to open the report file at {}: {}",
-            exp.report.display(),
-            e
-        ))
-    })?;
+pub fn run(mut exp: Export) -> Result<(), Box<dyn std::error::Error>> {
+    let cfg = meta::assemble_components(&mut exp.components)?;
+    let mut log_file = hopefully!(
+        File::open(&exp.report),
+        format!("failed to open the report file at {}", exp.report.display(),)
```
Comment (jonlamb-gh) [inline modality-probe-cli/src/export/mod.rs:64]: Minor nit: can we capitalize the error messages to match the rest of the subcommand error outputs.

File: modality-probe-cli/src/meta.rs (hunk 1 above)
Comment (jonlamb-gh) [inline modality-probe-cli/src/meta.rs:132]: Should we be doing `f32::from_bits(pl)` here?

**Review by jonlamb-gh — CHANGES_REQUESTED** [review]
Pretty neat!
A few small asks, otherwise looks good.
Also wondering if we should add some README content with examples of using `log` in this pr?

File: modality-probe-cli/src/meta.rs (hunk 1 above)
Comment (pittma) [reply modality-probe-cli/src/meta.rs:132]: Probably!

**Review by jonlamb-gh — APPROVED** [review]

File: modality-probe-cli/README.md [hunk 3]
```diff
@@ -45,6 +45,7 @@ SUBCOMMANDS:
 	export      	Export a collected trace as a Graphviz dot file
 	header-gen  	Generate Rust/C header files with event/probe id constants
 	help        	Prints this message or the help of the given subcommand(s)
+    log             Inspect a trace in the terminal as a log or an ASCII-based graph
```
Comment (jonlamb-gh) [inline modality-probe-cli/README.md:48]: I think the indentation got messed up here.

**Pull request merged**